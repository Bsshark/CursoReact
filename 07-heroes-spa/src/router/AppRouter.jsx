import { Routes, Route } from "react-router-dom"

import { HeroesRoutes } from "../heroes/"
import { LoginPage } from "../auth/pages/LoginPage"
import { PrivateRoute } from "./PrivateRoute";
import { Publicroute } from "./PublicRoute";

export const AppRouter = () => {
  return (
    <>
        <Routes>
            <Route path='login' element= {
              <Publicroute>
                <LoginPage />
              </Publicroute>
            } />
            
            <Route path="/*" element={
              <PrivateRoute>
                <HeroesRoutes />
              </PrivateRoute>
            } />
        </Routes>
    </>
  )
}
