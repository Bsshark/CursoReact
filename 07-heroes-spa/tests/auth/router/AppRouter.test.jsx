import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { AuthContext } from "../../../src/auth/context/AuthContext";
import { AppRouter } from "../../../src/router/AppRouter";

describe('Pruebas en AppRouter', () => { 
    it('should show login if not authenticated', () => {

        const contextValue = {
            logged: false
        }

        const { container } = render(
            <MemoryRouter initialEntries={['/marvel']}>
                <AuthContext.Provider value={contextValue}>
                    <AppRouter/>
                </AuthContext.Provider>
            </MemoryRouter>
        )

        expect(screen.getAllByText('Login').length).toBeGreaterThan(0);
        expect(container).toMatchSnapshot();
    });
    it('should show Marvel component if authenticated', () => {
        const contextValue = {
            logged: true,
            user: {
                id: 1,
                name: 'Abraham'
            }
        }

        const { container } = render(
            <MemoryRouter initialEntries={['/login']}>
                <AuthContext.Provider value={contextValue}>
                    <AppRouter/>
                </AuthContext.Provider>
            </MemoryRouter>
        )

        expect(container).toMatchSnapshot();
    });
 })