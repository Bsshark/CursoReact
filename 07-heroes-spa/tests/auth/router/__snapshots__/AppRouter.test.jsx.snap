// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AppRouter should show Marvel component if authenticated 1`] = `
<div>
  <nav
    class="navbar navbar-expand-sm navbar-dark bg-dark p-3"
  >
    <a
      class="navbar-brand"
      href="/"
    >
      Asociaciones
    </a>
    <div
      class="navbar-collapse"
    >
      <div
        class="navbar-nav"
      >
        <a
          aria-current="page"
          class="nav-link active"
          href="/marvel"
        >
          Marvel
        </a>
        <a
          class="nav-link "
          href="/dc"
        >
          DC
        </a>
        <a
          class="nav-link "
          href="/search"
        >
          Search
        </a>
      </div>
    </div>
    <div
      class="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        class="navbar-nav ml-auto"
      >
        <span
          class="nav-item nav-link text-primary"
        >
          Abraham
        </span>
        <button
          class="nav-item nav-link btn"
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>
  <div
    class="container"
  >
    <h1>
      DC Comics
    </h1>
    <hr />
    <div
      class="row rows-cols-1 row-cols-md-3 g-3"
    >
      <div
        class="col"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters animate__animated animate__fadeIn"
          >
            <div
              class="col-4"
            >
              <img
                alt="Spider Man"
                class="card-img"
                src="/assets/heroes/marvel-spider.jpg"
              />
            </div>
            <div
              class="col-8"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Spider Man
                </h5>
                <p
                  class="card-text"
                >
                   
                  Peter Parker
                   
                </p>
                <p
                  class="card-text"
                >
                  <small
                    class="text-muted"
                  >
                    Amazing Fantasy #15
                  </small>
                </p>
                <a
                  href="/hero/marvel-spider"
                >
                  Más info
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters animate__animated animate__fadeIn"
          >
            <div
              class="col-4"
            >
              <img
                alt="Captain America"
                class="card-img"
                src="/assets/heroes/marvel-captain.jpg"
              />
            </div>
            <div
              class="col-8"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Captain America
                </h5>
                <p
                  class="card-text"
                >
                   
                  Steve Rogers
                   
                </p>
                <p
                  class="card-text"
                >
                  <small
                    class="text-muted"
                  >
                    Captain America Comics #1
                  </small>
                </p>
                <a
                  href="/hero/marvel-captain"
                >
                  Más info
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters animate__animated animate__fadeIn"
          >
            <div
              class="col-4"
            >
              <img
                alt="Iron Man"
                class="card-img"
                src="/assets/heroes/marvel-iron.jpg"
              />
            </div>
            <div
              class="col-8"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Iron Man
                </h5>
                <p
                  class="card-text"
                >
                   
                  Tony Stark
                   
                </p>
                <p
                  class="card-text"
                >
                  <small
                    class="text-muted"
                  >
                    Tales of Suspense #39
                  </small>
                </p>
                <a
                  href="/hero/marvel-iron"
                >
                  Más info
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters animate__animated animate__fadeIn"
          >
            <div
              class="col-4"
            >
              <img
                alt="Thor"
                class="card-img"
                src="/assets/heroes/marvel-thor.jpg"
              />
            </div>
            <div
              class="col-8"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Thor
                </h5>
                <p
                  class="card-text"
                >
                   
                  Thor Odinson
                   
                </p>
                <p
                  class="card-text"
                >
                  <small
                    class="text-muted"
                  >
                    Journey into Myster #83
                  </small>
                </p>
                <a
                  href="/hero/marvel-thor"
                >
                  Más info
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters animate__animated animate__fadeIn"
          >
            <div
              class="col-4"
            >
              <img
                alt="Hulk"
                class="card-img"
                src="/assets/heroes/marvel-hulk.jpg"
              />
            </div>
            <div
              class="col-8"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Hulk
                </h5>
                <p
                  class="card-text"
                >
                   
                  Bruce Banner
                   
                </p>
                <p
                  class="card-text"
                >
                  <small
                    class="text-muted"
                  >
                    The Incredible Hulk #1
                  </small>
                </p>
                <a
                  href="/hero/marvel-hulk"
                >
                  Más info
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters animate__animated animate__fadeIn"
          >
            <div
              class="col-4"
            >
              <img
                alt="Wolverine"
                class="card-img"
                src="/assets/heroes/marvel-wolverine.jpg"
              />
            </div>
            <div
              class="col-8"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Wolverine
                </h5>
                <p
                  class="card-text"
                >
                   
                  James Howlett
                   
                </p>
                <p
                  class="card-text"
                >
                  <small
                    class="text-muted"
                  >
                    The Incredible Hulk #180
                  </small>
                </p>
                <a
                  href="/hero/marvel-wolverine"
                >
                  Más info
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters animate__animated animate__fadeIn"
          >
            <div
              class="col-4"
            >
              <img
                alt="Daredevil"
                class="card-img"
                src="/assets/heroes/marvel-daredevil.jpg"
              />
            </div>
            <div
              class="col-8"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Daredevil
                </h5>
                <p
                  class="card-text"
                >
                   
                  Matthew Michael Murdock
                   
                </p>
                <p
                  class="card-text"
                >
                  <small
                    class="text-muted"
                  >
                    Daredevil #1
                  </small>
                </p>
                <a
                  href="/hero/marvel-daredevil"
                >
                  Más info
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters animate__animated animate__fadeIn"
          >
            <div
              class="col-4"
            >
              <img
                alt="Hawkeye"
                class="card-img"
                src="/assets/heroes/marvel-hawkeye.jpg"
              />
            </div>
            <div
              class="col-8"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Hawkeye
                </h5>
                <p
                  class="card-text"
                >
                   
                  Clinton Francis Barton
                   
                </p>
                <p
                  class="card-text"
                >
                  <small
                    class="text-muted"
                  >
                    Tales of Suspense #57
                  </small>
                </p>
                <a
                  href="/hero/marvel-hawkeye"
                >
                  Más info
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters animate__animated animate__fadeIn"
          >
            <div
              class="col-4"
            >
              <img
                alt="Cyclops"
                class="card-img"
                src="/assets/heroes/marvel-cyclops.jpg"
              />
            </div>
            <div
              class="col-8"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Cyclops
                </h5>
                <p
                  class="card-text"
                >
                   
                  Scott Summers
                   
                </p>
                <p
                  class="card-text"
                >
                  <small
                    class="text-muted"
                  >
                    X-Men #1
                  </small>
                </p>
                <a
                  href="/hero/marvel-cyclops"
                >
                  Más info
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters animate__animated animate__fadeIn"
          >
            <div
              class="col-4"
            >
              <img
                alt="Silver Surfer"
                class="card-img"
                src="/assets/heroes/marvel-silver.jpg"
              />
            </div>
            <div
              class="col-8"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Silver Surfer
                </h5>
                <p
                  class="card-text"
                >
                   
                  Norrin Radd
                   
                </p>
                <p
                  class="card-text"
                >
                  <small
                    class="text-muted"
                  >
                    The Fantastic Four #48
                  </small>
                </p>
                <a
                  href="/hero/marvel-silver"
                >
                  Más info
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Pruebas en AppRouter should show login if not authenticated 1`] = `
<div>
  <div
    class="container mt-5"
  >
    <h1>
      Login
    </h1>
    <hr />
    <button
      class="btn btn-primary"
    >
      Login
    </button>
  </div>
</div>
`;
